src/oneD/Sim1D.cpp

```diff
@@ -447,22 +455,26 @@ void Sim1D::solve(int loglevel, bool refine_grid)
                 }
 
                 if (loglevel > 6) {
-                    save("debug_sim1d.yaml", "solution",
+                    header = "solution_" + std::to_string(attempt_counter) +"_NewtonSuccess";
```

Comment:
I'd suggest using `fmt` here, as we already use it quite extensively:
```suggestion
                    header = fmt::format("solution_{}_NewtonSuccess", attempt_counter);
```